runtime/interpreter/capability_controller.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/format"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+type CapabilityControllerValue struct {
```

Comment:
Maybe I am missing something, but where are these values actually constructed? It seems like the `NewCapabilityControllerValue` returns a `SimpleCompositeValue`, not a `CapabilityControllerValue`?

Reply:
ah yeah whoops, i originally implemented things in terms of `CapabilityControllerValue` but switched over to simple composite value. forgot to remove the actual struct definition and visitor code